eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.DetailNodeTreeStringPrinter;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.BlockCommentPosition;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * @codesnippet description should match naming pattern requirement below:
+ * <ol>
+ *   <li>Package, class and method names should be concatenated by dot '.'. Ex., packageName.className.methodName</li>
+ *   <li>Methods arguments should be concatenated by dash '-'. Ex. string-string  for methodName(String s, String s2)</li>
+ *   <li>Use '#' to concatenate 1) and 2), ex packageName.className.methodName#string-string</li>
+ * </ol>
+ *
+ */
+public class JavadocCodeSnippetCheck extends AbstractCheck {
+
+    private static final String CODE_SNIPPET_ANNOTATION = "@codesnippet";
+    private static final String MISSING_CODESNIPPET_TAG_MESSAGE = "Javadoc @codesnippet tag required description.";
```

Comment:
I'm not sure what this message is trying to say. 'Description' is meaningless to people - it is probably correct to say something like "There is a `@codesnippet` block in the JavaDoc, but it does not refer to any sample."